src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+        }
+    }
+}
+
+f32 func_80A9B244(BgHakuginPostUnkStruct* unkStruct) {
+    s32 i;
+    f32 ret = 0.0f;
+
+    for (i = 0; i < unkStruct->count; i++) {
+        ret += D_80A9D880[unkStruct->unk_0000[i].unk_00].unk_00;
+    }
+
+    return ret;
+}
+
+f32 func_80A9B2B8(BgHakuginPostUnkStruct* unkStruct) {
+    s32 i;
+
+    for (i = unkStruct->count - 1; i >= 0; i--) {
+        if ((unkStruct->unk_0000[i].unk_34 == 1) || (unkStruct->unk_0000[i].unk_34 == 2)) {
+            return unkStruct->unk_0000[i].unk_14.y + D_80A9D880[unkStruct->unk_0000[i].unk_00].unk_00;
+        }
+    }
+
+    return 0.0f;
+}
+
+BgHakuginPostUnkStruct1* func_80A9B32C(BgHakuginPostUnkStruct* unkStruct, BgHakuginPostUnkStruct1* unkStruct1) {
+    s32 i;
+
+    for (i = 0; i < unkStruct->count; i++) {
+        if (unkStruct1 == &unkStruct->unk_0000[i]) {
+            break;
+        }
+    }
+
+    i++;
+
+    if (i < unkStruct->count) {
+        return &unkStruct->unk_0000[i];
+    }
+
+    return NULL;
+}
+
+void func_80A9B384(Vec3f* arg0) {
+    MtxF* matrix = SysMatrix_GetCurrentState();
+
+    matrix->mf[3][0] = arg0->x;
+    matrix->mf[3][1] = arg0->y;
+    matrix->mf[3][2] = arg0->z;
+}
+
+void func_80A9B3BC(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp28 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor);
+    s32 sp24 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor) + 1;
+    s32 sp20;
+    s32 sp1C;
+
+    if (Flags_GetSwitch(globalCtx, sp28)) {
+        sp20 = true;
+    } else {
+        sp20 = false;
+    }
+
+    if (Flags_GetSwitch(globalCtx, sp24)) {
+        sp1C = true;
+    } else {
+        sp1C = false;
+    }
+
+    if (!(sp20 | sp1C)) {
+        Actor_SetSwitchFlag(globalCtx, sp28);
+        this->unk_170 = true;
+    } else {
+        this->unk_170 = sp20;
+    }
+
+    this->unk_174 = sp1C;
+}
+
+void func_80A9B46C(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp30 = BGHAKUGINPOST_GET_7F00(&this->dyna.actor);
+    s32 sp2C = BGHAKUGINPOST_GET_7F00(&this->dyna.actor) + 1;
+    s32 sp28;
+    s32 sp24;
+
+    if (Flags_GetSwitch(globalCtx, sp30) != 0) {
+        sp28 = true;
+    } else {
+        sp28 = false;
+    }
+
+    if (Flags_GetSwitch(globalCtx, sp2C) != 0) {
+        sp24 = true;
+    } else {
+        sp24 = false;
+    }
+
+    if ((this->unk_170 == false) && (sp28 == true)) {
+        Actor_UnsetSwitchFlag(globalCtx, sp2C);
+    } else if ((this->unk_174 == false) && (sp24 == true)) {
+        Actor_UnsetSwitchFlag(globalCtx, sp30);
+    }
+
+    this->unk_170 = sp28;
+    this->unk_174 = sp24;
+}
+
+void func_80A9B554(BgHakuginPost* this, GlobalContext* globalCtx, BgHakuginPostUnkStruct* unkStruct,
+                   BgHakuginPostUnkStruct1* unkStruct1) {
+    f32 spE4 = D_80A9D880[unkStruct1->unk_00].unk_00;
+    s32 i;
+    s16 val;
+    BgHakuginPostUnkStruct2* unkStruct2;
+    BgHakuginPostUnkStruct1* temp_unkStruct1;
```

Comment:
```suggestion
    BgHakuginPostUnkStruct1* unkStruct1Temp;
```
Or it can be any other combination of unk and Temp, just try to keep it to Camel case instead of snake_case